test_blob_autoload.c: -Werror=unused-function on builds without CONFIG_AI_SCHEDULER

\`make test\` (default QEMU virt, no AI scheduler) currently fails on \`main\` with:

\`\`\`
kernel/tests/test_blob_autoload.c:276:13: error: 'build_long_path' defined but not used [-Werror=unused-function]
kernel/tests/test_blob_autoload.c:110:15: error: 'build_eviction_mlp_payload' defined but not used [-Werror=unused-function]
\`\`\`

Both helpers are referenced only inside \`#ifdef CONFIG_AI_SCHEDULER\` blocks (line 507+). When \`AI_SCHED\` is not set at the make-target level (the default), the compiler can't see any users, so the strict-warnings build refuses.

Reproducer:
\`\`\`bash
make kernel-test-clean && make test
\`\`\`

Suggested fix: wrap the two function definitions in \`#ifdef CONFIG_AI_SCHEDULER\` to match their call sites. Two small \`#ifdef\` / \`#endif\` pairs around the function bodies. ~4 lines.

This blocks any PR that depends on a clean \`make test\` baseline (currently #395 — Stage 4 of the dynamic-kernel-replace plan). Filing as P1 since CI / local-test green-pass is broken on every default build.

--- a/kernel/tests/test_blob_autoload.c
+++ b/kernel/tests/test_blob_autoload.c
@@ -107,6 +107,9 @@ static size_t build_eviction_xgb_payload(uint8_t *out, size_t out_cap)
     return cursor;
 }
 
+/* Only used by the CONFIG_AI_SCHEDULER tests below; gate to silence
+ * `-Werror=unused-function` on default builds (issue #399). */
+#ifdef CONFIG_AI_SCHEDULER
 static size_t build_eviction_mlp_payload(uint32_t out_weight_bits,
                                          uint8_t *out,
                                          size_t out_cap)
@@ -165,6 +168,7 @@ static size_t build_eviction_mlp_payload(uint32_t out_weight_bits,
 
     return cursor;
 }
+#endif /* CONFIG_AI_SCHEDULER for build_eviction_mlp_payload */
 
 #ifdef CONFIG_AI_SCHEDULER
 static size_t build_sched_mlp_payload(uint32_t out_weight_bits,
@@ -273,6 +277,9 @@ static int read_text_file(const char *path, char *buf, size_t cap)
     return n;
 }
 
+/* Only used by the CONFIG_AI_SCHEDULER tests below; gate to silence
+ * `-Werror=unused-function` on default builds (issue #399). */
+#ifdef CONFIG_AI_SCHEDULER
 static void build_long_path(char *out, size_t cap,
                             const char *stem, char fill,
                             const char *suffix)
@@ -294,6 +301,7 @@ static void build_long_path(char *out, size_t cap,
     memcpy(out + prefix_len + stem_len + fill_len, suffix, suffix_len);
     out[target_len] = '\0';
 }
+#endif /* CONFIG_AI_SCHEDULER for build_long_path */
 
 static void test_blob_autoload_init_creates_conf(void)
 {

